src/RendezvousViewProvider.ts

```diff
@@ -0,0 +1,272 @@
+import * as arraySort from 'array-sort';
+import * as vscode from 'vscode';
+
+import { RendezvousHistory } from './RendezvousTracker';
+
+export class RendezvousViewProvider implements vscode.TreeDataProvider<vscode.TreeItem> {
+
+    constructor(context: vscode.ExtensionContext) {
+        this.enableSmartSorting();
+
+        // #region Register sorting commands
+        let subscriptions = context.subscriptions;
+        subscriptions.push(vscode.commands.registerCommand('extension.brightscript.rendezvous.toggleSortMethod', () => {
+            if (!this.enableSmartSorting()) {
```

Comment:
This is a little confusing, because the actions are also returning state. Can you try to separate the `enable/disable` functionality from the `is it enabled / disabled` for this condition? Something like:
```
this.enableSmartSorting();
if (this.smartSortingIsEnabled === false) {
    this.disableSmartSorting();
}
```

Also, can you add a comment here explaining why calling `enableSmartSorting` would do anything other than enable smart sorting, and why you even need to check? Perhaps the naming of the functions should be updated, like `tryEnableSmartSorting`, or break out the functionality so that enable actually always enables, and you do conditional checks around it instead.

Reply:
Changing this to a `toggleSmartSorting` function that will simply swap between the two.